src/App.tsx

```diff
@@ -1,14 +1,46 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedValue, setSelect] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    (async () => {
+      const todosLoaded = await getTodos();
+
+      setTodos(todosLoaded);
+      setLoading(false);
+    })();
+  }, []);
+
+  const includesQuery = (text:string) => {
+    return text.toLowerCase().includes(query.toLowerCase());
+  };
+
+  const visibleTodos = [...todos].filter((todo) => {
+    switch (selectedValue) {
+      case 'completed':
+        return todo.completed && includesQuery(todo.title);
+      case 'active':
+        return !todo.completed && includesQuery(todo.title);
+      default:
+        return includesQuery(todo.title);
+    }
+  });
```

Comment:
useMemo for preparing visibleTodos

Reply:
Hi. I used useMemo as required, but it works improperly, the list of todos is empty when first load. Maybe you have an idea what I did wrong? Please, check last commit. Thank you.

Reply:
it could be associated with inappropriate dependency array